modules/dataframe/include/inviwo/dataframe/datastructures/column.h

```diff
@@ -358,10 +362,19 @@ class IVW_MODULE_DATAFRAME_API CategoricalColumn : public TemplateColumn<std::ui
     /**
      * \brief returns column contents as list of categorical values
      *
-     * @return all categorical values stored in column
+     * @return all categorical values stored in the column
+     * @see values
      */
     std::vector<std::string> getValues() const;
 
+    /**
+     * returns column contents as iterator range of std::string_view
+     *
```

Comment:
you can add the explicit return types with the lambda if you want to... it is just a little bit of typing to get right...

Reply:
In principle we should have replaced the begin/end from the template column, but since we use inheritance to reuse code, that is not very difficult. I would have removed the template col inheritance. But that will probably break some code now...

Reply:
begin/end iterators are now returning `const std::string&`. Changing values is not supported as this would also affect the internal representation and possibly the lookup tables.

Reply:
The `values()` is not needed anymore?

Reply:
in principle not. There are however use cases since `values()` returns an iterator range of `begin()` and `end()`